server/src/main/java/io/druid/server/http/NamespacesResource.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.http;
+
+import com.fasterxml.jackson.jaxrs.smile.SmileMediaTypes;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.metamx.common.IAE;
+import com.metamx.common.RE;
+import com.metamx.common.logger.Logger;
+import io.druid.audit.AuditInfo;
+import io.druid.audit.AuditManager;
+import io.druid.common.utils.ServletResourceUtils;
+import io.druid.server.namespace.cache.NamespaceCoordinatorManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Contains information about namespaces
+ */
+@Path("/druid/coordinator/v1/namespaces")
+public class NamespacesResource
+{
+  private static final Logger log = new Logger(NamespacesResource.class);
+  private final NamespaceCoordinatorManager namespaceCoordinatorManager;
+
+  @Inject
+  public NamespacesResource(
+      final NamespaceCoordinatorManager namespaceCoordinatorManager
+  )
+  {
+    this.namespaceCoordinatorManager = namespaceCoordinatorManager;
+  }
+
+  @GET
+  @Produces({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response getNamespaces()
+  {
+    try {
+      final List<Map<String, Object>> knownNamespaces = namespaceCoordinatorManager.getKnownNamespaces();
```

Comment:
because the original design had the name as a required part of the object. 

After our conversation I'll be changing this.